dev/com.ibm.ws.jpa.tests.jpa_32_fat/test-applications/jakartadata/src/io/openliberty/jpa/data/tests/web/JakartaDataRecreateServlet.java

```diff
@@ -169,6 +169,70 @@ public void testOLGH28913() throws Exception {
 
         assertEquals(character.getHexadecimal(), result);
     }
+    
+    @Test
+    @Ignore("Reference issue:https://github.com/OpenLiberty/open-liberty/issues/29459")
+    public void testOLGH29459() throws Exception {
+        int x1 = 0, y1 = 0, x2 = 120, y2 = 209;
+        Segment segment = Segment.of(x1, y1, x2, y2);
+        List<Exception> exceptions = new ArrayList<>();
+        
+        if (tx.getStatus() != jakarta.transaction.Status.STATUS_ACTIVE) {
+            tx.begin();  
+        }
+
+        try {           
+            em.persist(segment);            
+            tx.commit();
+        } catch (Exception e) {            
+            if (tx.getStatus() == jakarta.transaction.Status.STATUS_ACTIVE) {
+                tx.rollback();
+            }
+            exceptions.add(e);
+        }
```

Comment:
praise: This is a valid time to check the transaction status. If `tx.commit()` threw an exception it could be a `RollbackException` or something else. If it was a `RollbackException` then there is no need to call `tx.rollback()`